context/llpcContext.h

```diff
@@ -203,32 +180,6 @@ class Context : public llvm::LLVMContext
     std::unique_ptr<llvm::TargetMachine> m_pTargetMachine; // Target machine
     bool                          m_scalarBlockLayout = false;  // scalarBlockLayout option from last pipeline compile
     bool                          m_robustBufferAccess = false; // robustBufferAccess option from last pipeline compile
-
-    // Pre-constructed LLVM types
-    struct
-    {
-        llvm::Type* pBoolTy;      // Bool
```

Comment:
I am OK to remove those basic types since builder can provide this. For the derived type, I think we can make some methods in LLPC builder in the future to provide them especially some of them are frequently used, such as <2 x float>, <3 x float>, <4 x float>, <4 x i32>, <8 x i32>. It is somwhaat inconvenient to use VectorType::get(...), The problem is that implementation within builder can use such methods while implementation in patch couldn't call them yet. I am still considering this.

Reply:
In a future commit, maybe we should introduce a new builder class in between IRBuilder and Builder, containing utility methods such as this. That class could then be used instead of IRBuilder in patch.